ddtrace/contrib/graphql/patch.py

```diff
@@ -0,0 +1,271 @@
+import os
+import re
+import sys
+from typing import TYPE_CHECKING
+
+import graphql
+from graphql import MiddlewareManager
+from graphql.error import GraphQLError
+from graphql.execution import ExecutionResult
+from graphql.language.source import Source
+
+from ddtrace import config
+from ddtrace.constants import ANALYTICS_SAMPLE_RATE_KEY
+from ddtrace.constants import ERROR_MSG
+from ddtrace.constants import ERROR_STACK
+from ddtrace.constants import ERROR_TYPE
+from ddtrace.constants import SPAN_MEASURED_KEY
+from ddtrace.internal.compat import stringify
+from ddtrace.internal.utils import ArgumentError
+from ddtrace.internal.utils import get_argument_value
+from ddtrace.internal.utils import set_argument_value
+from ddtrace.internal.utils.formats import asbool
+from ddtrace.internal.utils.version import parse_version
+from ddtrace.internal.wrapping import unwrap
+from ddtrace.internal.wrapping import wrap
+from ddtrace.pin import Pin
+
+from .. import trace_utils
+from ...ext import SpanTypes
+
+
+if TYPE_CHECKING:
+    from typing import Callable
+    from typing import Dict
+    from typing import Iterable
+    from typing import List
+    from typing import Tuple
+    from typing import Union
+
+    from ddtrace import Span
+
+
+_graphql_version_str = getattr(graphql, "__version__")
+graphql_version = parse_version(_graphql_version_str)
+
+if graphql_version < (3, 0):
+    from graphql.language.ast import Document
+else:
+    from graphql.language.ast import DocumentNode as Document
+
+config._add("graphql", dict(_default_service="graphql"))
+
+
+def _graphql_resolvers_enabled():
+    """
+    Patching graphql resolvers may produce large traces.
+    DD_TRACE_GRAPHQL_RESOLVERS_ENABLED is disabled by default.
+    """
+    return asbool(os.getenv("DD_TRACE_GRAPHQL_RESOLVERS_ENABLED", default=False))
```

Comment:
why not have this as part of the graphql config?

checking `os.getenv()` and calling `asbool` every time is going to have a performance hit.

e.g.

```
config._add(
    "graphql",
    dict(
        _default_service="graphql",
        resolvers_enabled=asbool(os.getenv("DD_TRACE_GRAPHQL_RESOLVERS_ENABLED", default=False)),
    ),
)
```

Then we can just check: `if config.graphql.resolvers_enabled`. This is more consistent with the way other integrations are setup, and allows someone to change this via programmatic config instead of env variables only.